File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> myMap;

        for (int i = 0; i < n; i++) {
            int val = nums[i];
            if (myMap[nums[i]] == 1) {
                return true;
            }
            myMap[nums[i]]++;
        }
        return false;
    }

};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        def backtracking(i, group):
            if sum(group) >= target or i == len(nums):
                if sum(group) == target: res.append(group.copy())
                return

            group.append(nums[i])
            backtracking(i, group)

            group.pop()
            backtracking(i+1, group)

        backtracking(0, [])
        return res
            
            